unstructured/ingest/connector/confluence.py

```diff
@@ -0,0 +1,164 @@
+            / self.file_meta.space_id
+            / f"{self.file_meta.document_id}.txt"
+        ).resolve()
+
+    @property
+    def _output_filename(self):
+        """Create output file path based on output directory, space id and document id"""
+        output_file = f"{self.file_meta.document_id}.json"
+        return Path(self.standard_config.output_dir) / self.file_meta.space_id / output_file
+
+    @requires_dependencies(["atlassian"])
+    @BaseIngestDoc.skip_if_file_exists
+    def get_file(self):
+        logger.debug(f"Fetching {self} - PID: {os.getpid()}")
+
+        # TODO: instead of having a separate connection object for each doc,
+        # have a separate connection object for each process
+        confluence = Confluence(
+            self.config.url,
+            username=self.config.user_email,
+            password=self.config.api_token,
+        )
+
+        result = confluence.get_page_by_id(page_id=self.file_meta.document_id, expand="body.view")
+        self.document = result["body"]["view"]["value"]
+        self.filename.parent.mkdir(parents=True, exist_ok=True)
+        with open(self.filename, "w", encoding="utf8") as f:
+            f.write(self.document)
+
+
+@requires_dependencies(["atlassian"])
+@dataclass
+class ConfluenceConnector(ConnectorCleanupMixin, BaseConnector):
+    """Fetches the body field from all documents from all spaces in a Confluence Cloud instance."""
+
+    config: SimpleConfluenceConfig
+
+    def __init__(
+        self,
+        standard_config: StandardConnectorConfig,
+        config: SimpleConfluenceConfig,
+    ):
+        super().__init__(standard_config, config)
+
+    @requires_dependencies(["atlassian"])
+    def initialize(self):
+        self.confluence = Confluence(
+            url=self.config.url,
+            username=self.config.user_email,
+            password=self.config.api_token,
+        )
+
+    @requires_dependencies(["atlassian"])
+    def _get_all_space_ids(self, start: int = 0, limit: int = 500, expand=None):
+        """Fetches all spaces in a confluence domain"""
+        results = self.confluence.get_all_spaces(start=start, limit=limit, expand=expand)
+        space_ids = [space["key"] for space in results["results"]]
+        return space_ids
+
+    @requires_dependencies(["atlassian"])
+    def _get_all_docs_ids_in_space(
+        self,
+        space_id: str,
+        start: int = 0,
+        limit: int = 100,
+        status=None,
+        expand=None,
+        content_type: str = "page",
+    ):
+        results = self.confluence.get_all_pages_from_space(
+            space_id,
+            start=start,
+            limit=limit,
+            status=status,
+            expand=expand,
+            content_type=content_type,
+        )
+        doc_ids = [(space_id, doc["id"]) for doc in results]
+        return doc_ids
+
+    @requires_dependencies(["atlassian"])
+    def _get_all_doc_ids_in_all_spaces(self):
```

Comment:
so we always return all spaces? should we allow the user to specify a space? either by name or id? and then maybe default to all spaces if they don't specify?

Reply:
I think a list of spaces makes sense then (in case they'd like a specific set of spaces)

Agreed on defaulting to all spaces if they don't specify 👍